ports/esp32/mphalport.h

```diff
@@ -48,6 +48,10 @@
 #define MP_TASK_COREID (0)
 #endif
 
+#if MICROPY_UNICORE
```

Comment:
Instead of using this, use the existing `CONFIG_FREERTOS_UNICORE` macro.

Reply:
I did use `CONFIG_FREERTOS_UNICORE` as well, see `sdkconfig.board`. That kept giving me assertion errors from FreeRTOS where this line in `mphalport.h` *still* tried to explicitly start tasks in Core 1. See https://forum.micropython.org/viewtopic.php?f=3&t=12055

Reply:
Ah, what I meant was to do the following:
```c
#if CONFIG_FREERTOS_UNICORE
#define MP_TASK_COREID (0)
#endif
```
Then you don't need to define a new `MICROPY_UNICORE` option.

But, actually, you might want to combine it with the above block, something like this:
```c
#if CONFIG_FREERTOS_UNICORE
#define MP_TASK_COREID (0)
#elif ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 2, 0)
#define MP_TASK_COREID (1)
#else
#define MP_TASK_COREID (0)
#endif
```

Reply:
I'm going to be a while on this. My test board died.